test/parallel/test-fs-read-position-validation.js

```diff
@@ -0,0 +1,126 @@
+    }
+  }));
+  try {
+    fdSync = fs.openSync(filepath, 'r');
+    fs.readSync(fdSync, buffer, offset, length, position);
+    fs.readSync(fdSync, buffer, { offset, length, position });
+  } finally {
+    if (fdSync) fs.closeSync(fdSync);
+  }
+  try {
+    fh = await fs.promises.open(filepath, 'r');
+    await fh.read(buffer, offset, length, position);
+    await fh.read({ buffer, offset, length, position });
+  } finally {
+    await fh?.close();
+  }
+}
+
+async function testInvalid(code, position, skipCb = false) {
+  let fdSync;
+  let fh;
+  try {
+    fdSync = fs.openSync(filepath, 'r');
+    assert.throws(
+      () => fs.readSync(fdSync, buffer, offset, length, position),
+      { code }
+    );
+    assert.throws(
+      () => fs.readSync(fdSync, buffer, { offset, length, position }),
+      { code }
+    );
+  } finally {
+    if (fdSync) fs.closeSync(fdSync);
+  }
+  try {
+    fh = await fs.promises.open(filepath, 'r');
+    assert.rejects(
+      fh.read(buffer, offset, length, position),
+      { code }
+    );
+    assert.rejects(
+      fh.read({ buffer, offset, length, position }),
+      { code }
+    );
+  } finally {
+    await fh?.close();
+  }
+
+  // Allows to skip Callback API test for uncatchable cases (for example, EINVAL)
+  if (skipCb) return;
+
+  fs.open(filepath, 'r', common.mustSucceed((fd) => {
+    try {
+      assert.throws(
+        () => fs.read(fd, buffer, offset, length, position, common.mustNotCall()),
+        { code }
+      );
+      assert.throws(
+        () => fs.read(fd, { buffer, offset, length, position }, common.mustNotCall()),
+        { code }
+      );
+    } finally {
+      fs.close(fd, common.mustSucceed());
+    }
+  }));
+}
+
+(async () => {
+  await testValid(undefined);
+  await testValid(null);
+  await testValid(-1);
+  await testValid(-1n);
+
+  await testValid(0);
+  await testValid(0n);
+  await testValid(1);
+  await testValid(1n);
+  await testValid(9);
+  await testValid(9n);
+  await testValid(Number.MAX_SAFE_INTEGER);
+
+  await testValid(2n ** 63n - 1n - BigInt(length));
+  await testInvalid('ERR_OUT_OF_RANGE', 2n ** 63n);
+  await testInvalid('ERR_OUT_OF_RANGE', 2n ** 63n - BigInt(length));
+
+  await testInvalid('ERR_OUT_OF_RANGE', NaN);
+  await testInvalid('ERR_OUT_OF_RANGE', -Infinity);
+  await testInvalid('ERR_OUT_OF_RANGE', Infinity);
+  await testInvalid('ERR_OUT_OF_RANGE', -2);
+  await testInvalid('ERR_OUT_OF_RANGE', -2n);
+  await testInvalid('ERR_OUT_OF_RANGE', -0.999);
+  await testInvalid('ERR_OUT_OF_RANGE', -(2n ** 64n));
+  await testInvalid('ERR_OUT_OF_RANGE', Number.MIN_SAFE_INTEGER);
+  await testInvalid('ERR_OUT_OF_RANGE', Number.MAX_SAFE_INTEGER + 1);
+  await testInvalid('ERR_OUT_OF_RANGE', Number.MAX_VALUE);
+
+  for (const badTypeValue of [
+    false, true, '1', Symbol(1), {}, [], () => {}, Promise.resolve(1),
+  ]) {
+    await testInvalid('ERR_INVALID_ARG_TYPE', badTypeValue);
+  }
+})()
+.then(common.mustCall());
```

Comment:
Alternatively you can use the `.mjs` extension and use TLA instead of an IIFE (to be clear both are perfectly fine, use whichever you prefer)

Reply:
Sure, why not. I do prefer ESM. :)